### PR TITLE
us003 - add capability to roles view

### PR DESCRIPTION
**Pull request by cdrainkainos**

### CODE REVIEW

File: src/main/java/com/kainos/ea/dao/RolesDao.java [hunk 1]
```diff
@@ -15,14 +15,20 @@ public RolesDao(){
...
+                    "JOIN capability " +
+                    "ON (job_family.capability_id = capability.id);");
 
             List<JobRole> jobRoles = new ArrayList<>();
 
             while (rs.next()) {
-                jobRoles.add(new JobRole(rs.getInt("id"), rs.getString("kainos_job_title")));
+                JobRole role = new JobRole(
+                        rs.getInt("id"),
+                        rs.getString("Kainos_Job_Title"),
```
Comment (shaunganley) [inline src/main/java/com/kainos/ea/dao/RolesDao.java]: looks like some inconsistencies in column names

File: src/test/java/com/kainos/ea/integration/WebServiceIntegrationTest.java [hunk 2]
```diff
@@ -30,4 +32,18 @@ void getEmployees_shouldReturnListOfEmployees() {
...
+        ObjectMapper mapper = new ObjectMapper();
+        JsonNode response = APP.client().target("http://localhost:8080/api/job-roles")
+                .request()
+                .get(JsonNode.class);
+        List<JobRole> JobRoleList = mapper.convertValue(
+                response,
+                new TypeReference<List<JobRole>>(){}
+        );
+
+        assertNotNull(JobRoleList.get(0).getCapability_name());
```
Comment (shaunganley) [inline src/test/java/com/kainos/ea/integration/WebServiceIntegrationTest.java:42]: can you add this check to an existing integration test so we reduce the amount of HTTP calls needing to be made

File: src/test/java/com/kainos/ea/service/RolesServiceTest.java [hunk 3]
```diff
@@ -5,22 +5,33 @@
...
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.*;
 
 class RolesServiceTest {
 
+    static final DropwizardAppExtension<trueConfiguration> APP = new DropwizardAppExtension<>(
+            trueApplication.class, null,
+            new ResourceConfigurationSourceProvider()
+    );
```
Comment (shaunganley) [inline src/test/java/com/kainos/ea/service/RolesServiceTest.java]: this shouldn't be in a unit test class, this is for integration tests

**Review by shaunganley — CHANGES_REQUESTED** [review]

File: src/main/java/com/kainos/ea/dao/RolesDao.java (hunk 1 above)
Comment (cdrainkainos) [reply src/main/java/com/kainos/ea/dao/RolesDao.java]: names fixed

File: src/test/java/com/kainos/ea/integration/WebServiceIntegrationTest.java (hunk 2 above)
Comment (cdrainkainos) [reply src/test/java/com/kainos/ea/integration/WebServiceIntegrationTest.java:42]: added not null checks for each field to existing test

File: src/test/java/com/kainos/ea/service/RolesServiceTest.java (hunk 3 above)
Comment (cdrainkainos) [reply src/test/java/com/kainos/ea/service/RolesServiceTest.java]: removed

**Review by shaunganley — APPROVED** [review]

**Review by bannister-samuel — APPROVED** [review]

**Pull request merged**